opm/simulators/linalg/bda/BdaBridge.cpp

```diff
@@ -168,27 +169,28 @@ int checkZeroDiagonal(BridgeMatrix& mat) {
 // iterate sparsity pattern from Matrix and put colIndices and rowPointers in arrays
 // sparsity pattern should stay the same
 // this could be removed if Dune::BCRSMatrix features an API call that returns colIndices and rowPointers
-template <class BridgeMatrix>
-void getSparsityPattern(BridgeMatrix& mat, std::vector<int> &h_rows, std::vector<int> &h_cols) {
+template <class BridgeMatrix, class BridgeVector, int block_size>
+void BdaBridge<BridgeMatrix, BridgeVector, block_size>::getSparsityPattern(const BridgeMatrix& mat, std::vector<int> &h_rows, std::vector<int> &h_cols) {
     int sum_nnzs = 0;
 
-    // convert colIndices and rowPointers
-    if (h_rows.empty()) {
-        h_rows.emplace_back(0);
-            for (typename BridgeMatrix::const_iterator r = mat.begin(); r != mat.end(); ++r) {
-                int size_row = 0;
-                for (auto c = r->begin(); c != r->end(); ++c) {
-                    h_cols.emplace_back(c.index());
-                    size_row++;
-                }
-                sum_nnzs += size_row;
-                h_rows.emplace_back(sum_nnzs);
-            }
+    h_rows.clear();
+    h_cols.clear();
 
-        // h_rows and h_cols could be changed to 'unsigned int', but cusparse expects 'int'
-        if (static_cast<unsigned int>(h_rows[mat.N()]) != mat.nonzeroes()) {
-            OPM_THROW(std::logic_error, "Error size of rows do not sum to number of nonzeroes in BdaBridge::getSparsityPattern()");
+    // convert colIndices and rowPointers
+    h_rows.emplace_back(0);
+    for (typename BridgeMatrix::const_iterator r = mat.begin(); r != mat.end(); ++r) {
+        int size_row = 0;
+        for (auto c = r->begin(); c != r->end(); ++c) {
+            h_cols.emplace_back(c.index());
+            size_row++;
         }
+        sum_nnzs += size_row;
+        h_rows.emplace_back(sum_nnzs);
```

Comment:
Probably a matter of taste, but alas: One could skip the counters `size_row` (not really used) and `sum_nnz` (`==h_cols.size()`).

Reply:
afb25e9